Global.h

```diff
@@ -698,3 +698,16 @@ namespace vstd
 }
 using vstd::operator-=;
 using vstd::make_unique;
+
+#if defined VCMI_ANDROID && defined NO_STD_TOSTRING

```

Comment:
Also, I added VCMI_ANDROID guard there (and most of other places), because, for now, I tried to change desktop's version logic as little as possible.